compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -605,7 +605,7 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
             if (tparams1.nonEmpty)
               return recur(tp1.EtaExpand(tparams1), tp2) || fourthTry
             tp2 match {
-              case EtaExpansion(tycon2) if tycon2.symbol.isClass && tycon2.symbol.is(JavaDefined) =>
+              case EtaExpansion(tycon2: TypeRef) if tycon2.typeSymbol.isClass && tycon2.symbol.is(JavaDefined) =>
```

Comment:
typeSymbol = symbol on a TypeRef, so you can use just symbol.
```suggestion
              case EtaExpansion(tycon2: TypeRef) if tycon2.symbol.isClass && tycon2.symbol.is(JavaDefined) =>
```